switchio/loops/proactor.py

```diff
@@ -0,0 +1,330 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2017 Tyler Goodlet <tgoodlet@gmail.com>
+"""
+``asyncio`` based proactor loop.
+"""
+import functools
+import itertools
+import asyncio
+import time
+import traceback
+from collections import deque
+from threading import Thread, current_thread, get_ident
+from .reactor import EventLoop
+from .. import utils
+from ..utils import get_event_time
+
+
+@asyncio.coroutine
+def just_yield():
+    """A "just yield" coroutine which triggers an interation of the event loop.
+
+    If you think this is a nightmare to understand have you asked yourself how
+    this will ever work once these legacy types of generator "coroutines" are
+    removed from the language?
+    """
+    yield
```

Comment:
The approach I took in some test code that needed to do this:

```python
release = functools.partial(asyncio.sleep, 0)
```

Reply:
Yeah i want to make it obvious that this is a ridiculous thing to expect users of `asyncio` to do.
I'd like to possibly talk to the cpython guys and get a proper future api for this.